topology/server.go

```diff
@@ -110,7 +115,27 @@ func (s *Server) RegisterRpcEndpoints() {
 }
 
 func (s *Server) ListenAndServe() {
-	http.ListenAndServe(s.Addr+":"+strconv.FormatInt(int64(s.Port), 10), s.Router)
+	defer s.wg.Done()
+	s.wg.Add(1)
+
+	listener, err := net.Listen("tcp", fmt.Sprintf("%s:%d", s.Addr, s.Port))
+	if err != nil {
+		panic(err)
```

Comment:
don't use panic, use logging.GetLogger().Fatal() with a simple and different message